src/UmbracoExamine.PDF/PdfLuceneIndex.cs

```diff
@@ -31,6 +33,62 @@ public PdfLuceneIndex(ILoggerFactory loggerFactory, string name, IOptionsMonitor
 
         #endregion
 
+        /// <summary>
+        ///     Special check for invalid paths
+        /// </summary>
+        /// <param name="values"></param>
+        /// <param name="onComplete"></param>
+        protected override void PerformIndexItems(IEnumerable<ValueSet> values, Action<IndexOperationEventArgs> onComplete)
+        {
+            // We don't want to re-enumerate this list, but we need to split it into 2x enumerables: invalid and valid items.
+            // The Invalid items will be deleted, these are items that have invalid paths (i.e. moved to the recycle bin, etc...)
+            // Then we'll index the Value group all together.
+            var invalidOrValid = values.GroupBy(v =>
+            {
+                if (!v.Values.TryGetValue("path", out IReadOnlyList<object> paths) || paths.Count <= 0 || paths[0] == null)
+                {
+                    return ValueSetValidationStatus.Failed;
+                }
+
+                ValueSetValidationResult validationResult = ValueSetValidator.Validate(v);
+
+                return validationResult.Status;
+            }).ToArray();
+
+            var hasDeletes = false;
+            var hasUpdates = false;
 
+            // ordering by descending so that Filtered/Failed processes first
+            foreach (IGrouping<ValueSetValidationStatus, ValueSet> group in invalidOrValid.OrderByDescending(x => x.Key))
+            {
+                switch (group.Key)
+                {
+                    case ValueSetValidationStatus.Valid:
+                        hasUpdates = true;
+
+                        //these are the valid ones, so just index them all at once
+                        base.PerformIndexItems(group.ToArray(), onComplete);
+                        break;
+                    case ValueSetValidationStatus.Failed:
+                        // don't index anything that is invalid
+                        break;
+                    case ValueSetValidationStatus.Filtered:
+                        hasDeletes = true;
+
+                        // these are the invalid/filtered items so we'll delete them
+                        // since the path is not valid we need to delete this item in
+                        // case it exists in the index already and has now
+                        // been moved to an invalid parent.
+                        base.PerformDeleteFromIndex(group.Select(x => x.Id), null);
+                        break;
+                }
+            }
+
+            if ((hasDeletes && !hasUpdates) || (!hasDeletes && !hasUpdates))
```

Comment:
Better late than never 🙈 

I do not get this line, it must be an error. Right now we could as well just check `if (!hasUpdates)`

Reply:
Hi @bergmania.  Most of that code was copied straight from the UmbracoContentIndex class so I just copied most of the code verbatim.  The same still exists in Umbraco here:

https://github.com/umbraco/Umbraco-CMS/blob/a72ffc8542d0ea5f44870a810c087dc9d3ccb9b9/src/Umbraco.Examine.Lucene/UmbracoContentIndex.cs#L111

I don't see why it needs to check `hasDeletes` as just calling `if (!hasUpdates)` will do the same thing.  I can look at amending the PR.